contracts/cw20-ics20/src/contract.rs

```diff
@@ -124,6 +143,48 @@ pub fn execute_transfer(
     Ok(res)
 }
 
+/// The gov contract can allow new contracts, or increase the gas limit on existing contracts.
+/// It cannot block or reduce the limit to avoid forcible sticking tokens in the channel.
+pub fn execute_allow(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    allow: AllowMsg,
+) -> Result<Response, ContractError> {
+    let cfg = CONFIG.load(deps.storage)?;
+    ensure_eq!(info.sender, cfg.gov_contract, ContractError::Unauthorized);
+
+    let contract = deps.api.addr_validate(&allow.contract)?;
+    let set = AllowInfo {
+        gas_limit: allow.gas_limit,
+    };
+    ALLOW_LIST.update(deps.storage, &contract, |old| {
+        if let Some(old) = old {
+            // we must ensure it increases the limit
+            match (old.gas_limit, set.gas_limit) {
+                (None, Some(_)) => return Err(ContractError::CannotLowerGas),
```

Comment:
If I read this condition correctly - there was no gas limit, now it is - then:
> ensure it increases the limit

it looks like it increases the limit (from 0 to anything above)?

Reply:
None means "infinite" - no limit
Some(x) means we apply a limit.

Probably should add a doc string here https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/results/submessages.rs#L36

But seeing that None is the default, so one can imagine no limit is applied.

Reply:
I read it as `None` being "no-limit", so infinite limit, so any limitation is decreasing limit, so its illegal. "Zero" limit doesn't make any sense even remotely, so if `None` would be "zero" it could be just never be an optional as it is never a value you want to keep.